dataprofiler/data_readers/parquet_data.py

```diff
@@ -128,11 +150,19 @@ def is_match(cls, file_path, options=None):
 
         # return to original position in stream
         if data_utils.is_stream_buffer(file_path):
+            file_path = cast(
+                Union[StringIO, BytesIO], file_path
+            )  # guaranteed by is_stream_buffer
```

Comment:
let's see if we can get rid of these `cast` statements

Reply:
Done.